docs/performance-testing.md

```diff
@@ -0,0 +1,89 @@
+# Performance Testing
+
+TEAMMATES makes use of [JMeter](https://jmeter.apache.org/) for performance testing.
+The relevant files are located in the `src/jmeter/` directory. The JMeter tests are the `.jmx` files located in the `src/jmeter/tests/` directory.
+
+TEAMMATES uses the [`jmeter-gradle-plugin`](https://github.com/jmeter-gradle-plugin/jmeter-gradle-plugin) for automating the processes of running performance tests and generating reports.
+
+## Performance Test Workflow
+
+Before running a performance test, you need to setup the relevant data in the GAE datastore. 
+To facilitate this, there are Java scripts in the `teammates.performance.scripts` package (`src/jmeter/java/teammates/performance/scripts/` directory).
+
+There are 3 steps involved in doing a performance test:
+
+1. Setting up the data: This is done by executing the relevant Java script in the `setup` package (e.g. `SetupStudentProfileTest`).
+    Executing the scrip will:
+    1. Create the relevant TEAMMATES JSON data for the datastore. It is stored as `src/jmeter/resources/data/testName.json`.
+    1. Create the relevant CSV Config data for the JMeter test plan. It is stored as `src/jmeter/resources/data/testNameConfig.csv`.
+    1. Create the entities in the datastore using the JSON data. This can be verified by checking your [local datastore](http://localhost:8080/_ah/admin/datastore).
+        > The backend server (`localhost:8080`) must be started for this to happen successfully.
+1. [Run the performance test](#running-performance-tests) specified in the JMeter `.jmx` file in the `src/jmeter/tests/` directory.
+1. Remove the data that was added to the datastore (optional, but recommended). To do so, execute the relevant Java script in the `teardown` package (e.g. `DeleteStudentProfileTestData`).
+   In addition, you can choose to delete the `.json` and `.csv` files that were generated in step 1. 
+
+## Running Performance Tests
+
+Start the backend server (`localhost:8080`) before executing the JMeter performance tests.
+
+### Using Gradle
+
+To run the JMeter tests, execute this command from the main project directory:
+```
+./gradlew clean jmRun
+```
+
+If it runs without any errors:
+- A log file is generated as `build/jmeter/jmeter.log`.
```

Comment:
![image](https://user-images.githubusercontent.com/28522090/54875038-f3abd700-4e31-11e9-9673-1f314f65aeee.png)
It is not normal to see these error messages in the log file right?

Reply:
@jacoblipech Did you set up the data before running the test?